nilearn/datasets/atlas.py

```diff
@@ -20,6 +20,103 @@
 _TALAIRACH_LEVELS = ['hemisphere', 'lobe', 'gyrus', 'tissue', 'ba']
 
 
+def fetch_difumo(dimension=64, resolution_mm=2, data_dir=None, resume=True, verbose=1):
+    """Fetch DiFuMo brain atlas
+
+    Direct download links from OSF:
+
+    dic = {64: https://osf.io/wjum7/download,
+           128: https://osf.io/kdfg3/download,
+           256: https://osf.io/vza2y/download,
+           512: https://osf.io/a23gw/download,
+           1024: https://osf.io/jpdum/download,
+           }
+
+    Parameters
+    ----------
+    dimension : int
+        Number of dimensions in the dictionary. Valid resolutions
+        available are {64, 128, 256, 512, 1024}.
+
+    resolution_mm : int
+        The resolution in mm of the atlas to fetch. Valid options
+        available are {2, 3}.
+
+    data_dir : string, optional
+        Path where data should be downloaded. By default,
+        files are downloaded in home directory.
+
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        Dictionary-like object, the interest attributes are :
+
+        - 'maps': str, 4D path to nifti file containing regions definition.
+        - 'labels': string list containing the labels of the regions.
+
+    References
+    ----------
+    Dadi, K., Varoquaux, G., Machlouzarides-Shalit, A., Gorgolewski, KJ.,
+    Wassermann, D., Thirion, B., Mensch, A.
+    Fine-grain atlases of functional modes for fMRI analysis,
+    Paper in preparation
+    """
+    dic = {64: 'wjum7',
+           128: 'kdfg3',
+           256: 'vza2y',
+           512: 'a23gw',
+           1024: 'jpdum',
+           }
+    valid_dimensions = [64, 128, 256, 512, 1024]
+    valid_resolution_mm = [2, 3]
+    if dimension not in valid_dimensions:
+        raise ValueError("Requested dimension={} is not available. Valid "
+                         "options: {}".format(dimension, valid_dimensions))
+    if resolution_mm not in valid_resolution_mm:
+        raise ValueError("Requested resolution_mm={} is not available. Valid "
+                         "options: {}".format(resolution_mm,
+                                              valid_resolution_mm))
+    
+    url = 'https://osf.io/{}/download'.format(dic[dimension])
+    opts = {'uncompress': True}
+
+    csv_file = os.path.join('{0}', 'labels_{0}_dictionary.csv')
+    if resolution_mm != 3:
+        nifti_file = os.path.join('{0}', 'maps.nii.gz')
+    else:
+        nifti_file = os.path.join('{0}', '3mm', 'resampled_maps.nii.gz')
+
+    files = [(csv_file.format(dimension), url, opts),
+             (nifti_file.format(dimension), url, opts)]
+
+    dataset_name = 'difumo_atlases'
+
+    data_dir = _get_dataset_dir(dataset_name=dataset_name, data_dir=data_dir,
+                                verbose=verbose)
+
+    # Download the zip file, first
+    files_ = _fetch_files(data_dir, files, verbose=verbose)
+    labels = np.recfromcsv(files_[0])
```

Comment:
ahh ok. I don't reproduce the issue with dimension=128 because I may have checked with pandas not numpy. My bad.

Reply:
Yes, Pandas is clever enough to do it right. I went for Numpy because most fetchers do it this way (I think Pandas is not even imported in this file). 
To remove this bug, we can either (sorted by my preference):

- remove the extra commas directly on OSF (makes sense to me since this is a mistake in the data)
- use Pandas instead of Numpy to read the label CSV files
- remove the extra commas in the fetcher code (really dislike this one...)